lang/java/avro/src/main/java/org/apache/avro/generic/GenericData.java

```diff
@@ -594,7 +594,7 @@ public Object getField(Object record, String name, int position) {
   /** Produce state for repeated calls to {@link
    * #getField(Object,String,int,Object)} and {@link
    * #setField(Object,String,int,Object,Object)} on the same record.*/
-  protected Object getRecordState(Object record, Schema schema) { return null; }
+  public Object getRecordState(Object record, Schema schema) { return null; }
```

Comment:
With the recent writer change, I don't think this needs to be exposed any longer.